### DISCUSSION TITLE
Offsetting a shell to create a solid

### DISCUSSION THREAD

**Discussion by pruj-speed** [CadQuery/cadquery#1737] [Q&A]
Hi,

I've been using cadquery to generate 3D geometry from simple 2D shapes. In my pipeline I have been finding the intersection between two solid objects and then creating a shell out of the faces that are intersecting above a certain Z-height. Basically here is how my shell looks like. Its composed of 112 faces and is of type cadquery.occ_impl.shapes.Shell.

View 1
![image](https://github.com/user-attachments/assets/3752d3d9-334a-4f45-8e3f-211518327792)

View 2
![image](https://github.com/user-attachments/assets/f134ef24-737d-44e6-86a7-29dcb7aa9574)

I'm uploading the shell BREP here.
[offset_failing_shell.zip](https://github.com/user-attachments/files/18278150/offset_failing_shell.zip)

I'm using the offset function in shapes to convert this into a solid.

``` python
from cadquery.occ_impl.shapes import offset
from cadquery import importers
from cadquery.vis import show
shell = importers.importBrep('offset_failing_shell.BREP')
shell.val().isValid() # returns True
extruded_solid = offset(shell.val(),0.5E-3)
show(extruded_solid)
extruded_solid.isValid() # returns False
```

 However, the resulting solid looks funky.

![image](https://github.com/user-attachments/assets/dc3f0bcb-71f9-4394-8d9f-7f3d6c9d7357)

![image](https://github.com/user-attachments/assets/bae98dc9-2713-4c74-8cd0-f9b421cb2991)

My main goal is to mimick a deposition process where all exposed surfaces get thickened by a certain amount. Any advice on if this is the right approach to create such a thickened solid out of several faces? It seems to work fine for most cases. But not for this specific geometry.

**Comment by pruj-speed** [comment]
Hey @jmwright pinging you for some advice on offsetting shells. Are there any rules as to when a shell offset will fail? Any pointers will help me in debugging or changing my approach with this shell I'm working with.

**Comment by jmwright** [comment]
@pruj-speed It's hard to say, but if the shell was valid before the offset, it looks like a B-rep geometry glitch to me. You could also try displaying the offset shell in CQ-editor, and exporting it to STEP and/or STL to see if the glitch is persistent and not just a cadquery.vis.show VTK thing. 

Other than that, you could wait until [this PR](https://github.com/CadQuery/cadquery/pull/1589) is merged and then try the offset again with OCCT/OCP 7.8.1. You may need to use an Anaconda environment for your test since the release of the 7.8.1 wheel will probably lag the conda pacakge a bit. You will be on the bleeding edge of CadQuery/OCP/OCCT, but I do not know if it will fix your problem.

**Reply by pruj-speed** [reply]
I checked and the shell is valid before offset. The solid from offset looks funky in both the cq-editor and also in solidworks,edrawings when I view the step. I will keep an eye on this PR and try it out.
